### PR TITLE
Add settlement explanation

### PR DESCRIPTION
**Pull request by karlb**
The settlement was only explained in the diagram, which does not
convey information like timing differences, links to functions,
purpose, etc. Since the settlement is an essential part of Raiden,
having some explanatory text makes a lot of sense.

The coop settle is not enabled right now, but I consider it to be part
of the spec, nevertheless.

### CODE REVIEW

File: smart_contracts.rst [hunk 1]
```diff
@@ -524,7 +524,7 @@ Allows a channel participant to withdraw tokens from a channel without closing i
 Close a channel
 ^^^^^^^^^^^^^^^
 
-Allows anybody to close a channel with a channel participant's signature. After a channel is closed, the channel cannot be settled before the challenge period has ended.
+Allows anybody to close a channel with a channel participant's signature. Closing a channel initiates an uncooperative settlement and marks the start of the challenge period. Once challenge period has ended, the channel can be settled.
```
Comment (ezdac) [inline smart_contracts.rst:527]: Nitpick: Once *the* challenge period ...

File: smart_contracts.rst [hunk 2]
```diff
@@ -815,6 +815,16 @@ Opened Channel Lifecycle
 Channel Settlement
 ------------------
 
+If channel participants want to retrieve all tokens from a channel, they can do this by settling the channel. In case the other participant cooperates, this can be done quickly in a single transaction. If the other participant is not available or unwilling to cooperate, the tokens can still be retrieved through an uncooperative settlement. This process is slower and incurs higher gas fees, but is guaranteed to succeed within a limited time frame.
+
+The uncooperative settlement consists of three steps:
+
+* The :ref:`channel is closed <close-channel>` by (or on behalf of) one participant. This starts the challenge period.
+* The non-closing participant can :ref:`update the channel state <update-channel>` during the challenge period to ensure that all transfers he received are considered during settlement.
+* Once the channel period is over, either party can finally :ref:`settle the channel <settle-channel>` which will cause the TokenNetwork contract to return all unlocked tokens.
```
Comment (ezdac) [inline smart_contracts.rst:824]: Once the *challenge* period is over ...

**Review by ezdac — APPROVED** [review]
Looks good, nice and clear explanation.
Found 2 typos.

**Comment by karlb** [comment]
Thanks for spotting the typos! The ones not caught by the spell checker are always the harder ones.

**Pull request merged**